controllers/taskController.js

```diff
@@ -33,28 +17,26 @@ const getByID = (req, res) => {
 };
 
 const create = (req, res) => {
-    const title = req.body.title;
-    const course = req.body.course;
-    const deadline = req.body.deadline;
-    let task = {
-        id: tasks.length+1,
-        title: title,
-        course: course,
-        deadline: deadline
-    }
-    tasks.push(task);
+    let props = {
+        title: req.body.title,
+        course: req.body.course,
+        deadline: req.body.deadline
+    };
+    task = taskRepo.newTask(props);
```

Comment:
I think this makes a global variable because it's not declared in scope. It might help to setup a linter for stuff like this.